pkg/apis/v1/nodeclaim_conversion.go

```diff
@@ -93,8 +93,11 @@ func (in *NodeClaim) ConvertFrom(ctx context.Context, from apis.Convertible) err
 	if err != nil {
 		return err
 	}
-	in.Annotations = lo.Assign(in.Annotations, map[string]string{KubeletCompatabilityAnnotationKey: kubeletAnnotation})
-
+	if kubeletAnnotation == "" {
+		in.Annotations = lo.OmitByKeys(in.Annotations, []string{KubeletCompatibilityAnnotationKey})
```

Comment:
is this the same as `delete(in.Annotations, KubeletCompatibilityAnnotationKey)`?

Reply:
Yes